rdmd.d

```diff
@@ -59,6 +59,15 @@ version(unittest) {} else
 int main(string[] args)
 {
     //writeln("Invoked with: ", args);
+    string binPath = thisExePath();
+    if (binPath.exists && binPath.isFile && filenameCmp(baseName(binPath),
+                                                        baseName(args[0])) == 0)
```

Comment:
Not sure what you mean about the approach, but `filenameCmp` claims to be able to handle case-sensitivity.